pkg/receive/handler.go

```diff
@@ -339,6 +340,51 @@ func (h *Handler) receiveHTTP(w http.ResponseWriter, r *http.Request) {
 		tenant = h.options.DefaultTenantID
 	}
 
+	// Checking certs for possible value.
```

Comment:
I'd say this whole block and the 5 lines above should all be extracted into their own function like
`func (h *Handler) findTenant(r *http.Request) str`